acceptance/tests/puppet_device_test.rb

```diff
@@ -1,24 +1,23 @@
 test_name "puppet device is able to run and configure a node"
 
+tag 'server'
+
 teardown do
-  on(master, "[ -f /etc/puppetlabs/puppet/puppet.conf.bak ] && mv /etc/puppetlabs/puppet/puppet.conf.bak /etc/puppetlabs/puppet/puppet.conf")
   # revert permission changes to reset state for other tests
   on(master, "puppet agent -t --server #{master}")
 end
 
-# set sensible defaults
-on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
+step "Configure master for device runs"
 on(master, "puppet config set server $(hostname --fqdn)")
 
 codedir = "#{master.puppet['codedir']}/modules"
 confdir = master.puppet['confdir']
-common_options="--config /tmp/puppet.conf --debug --trace -w0"
 
 on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
 on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
 on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
```

Comment:
The `server`, `user`, and modules persist after this test finishes, which is bad because subsequent agent tests pluginsync all of the device modules, eg

```
  * Check in as gmp0lo62urp2fp1.delivery.puppetlabs.net-extensions

  gmp0lo62urp2fp1.delivery.puppetlabs.net (gmp0lo62urp2fp1.delivery.puppetlabs.net) 10:15:18$ env FACTER_test_dir="/tmp/certificate_extensions.hcU9LZ/" FACTER_TEST_DIR="/tmp/certificate_extensions.hcU9LZ/" puppet agent --test --waitforcert 0 --csr_attributes /tmp/certificate_extensions.hcU9LZ/csr_attributes.yaml --certname gmp0lo62urp2fp1.delivery.puppetlabs.net-extensions --ssldir /tmp/certificate_extensions.hcU9LZ/ssldir --server uohriyc80dtav60.delivery.puppetlabs.net
...
    Notice: /File[/opt/puppetlabs/puppet/cache/lib/puppet/util]/ensure: created
    Notice: /File[/opt/puppetlabs/puppet/cache/lib/puppet/util/network_device]/ensure: created
    Notice: /File[/opt/puppetlabs/puppet/cache/lib/puppet/util/network_device/spinner]/ensure: created
    Notice: /File[/opt/puppetlabs/puppet/cache/lib/puppet/util/network_device/spinner/device.rb]/ensure: defined content as '{md5}9121c2c43faf3acaedd654a3ec590513'
```

In cases where the master state changes like this, we usually create tempdir(s) and pass them to `with_puppet_running_on`. Something like:

```ruby
confdir = create_tmpdir_for_user(master, 'confdir')
codedir = create_tmpdir_for_user(master, 'codedir')

manifest = <<-MANIFEST
  File {
    ensure => directory,
    owner => #{master.puppet['user']},
    group => #{master.puppet['group']},
    mode => "0750",
  }
  file { "#{confdir}": }
  file { "#{codedir}": }
MANIFEST

apply_manifest_on(master, manifest, :catch_failures => true)

on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))

master_opts = {
  'main' => {
    'codedir' => "#{codedir}"
  }
}

...

on(master, "puppet config set user test --section main --confdir #{confdir}")

with_puppet_running_on(master, master_opts, confdir) do
  agents.each do |agent|
    ..
  end
end
```